javaslang/src/main/java/javaslang/collection/Arrays.java

```diff
@@ -27,6 +27,30 @@
         return arrayCopy(length, array, from, 0, length);
     }
 
+    /** Repeatedly group an array into equal sized sub-trees */
+    static Object[] grouped(Object[] array, int groupSize) {
+        assert array.length > groupSize;
+        final Object[] results = copy(empty(), 1 + ((array.length - 1) / groupSize));
+        results[0] = copyRange(array, 0, groupSize);
+
+        for (int start = groupSize, i = 1; start < array.length; i++) {
+            final int nextLength = Math.min(groupSize, array.length - (i * groupSize));
```

Comment:
it should be the same as for `Traversable`'s `groupes`, though it's applied recursively until the root array is small enough (<=32)